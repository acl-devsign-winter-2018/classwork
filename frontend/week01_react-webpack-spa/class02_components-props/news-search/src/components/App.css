header, main {
  padding: 20px;
}