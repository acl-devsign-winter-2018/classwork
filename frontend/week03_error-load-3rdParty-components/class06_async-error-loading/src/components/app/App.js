import React, { Component } from 'react';
import { connect } from 'react-redux';
import Notes from '../notes/Notes';

class App extends Component {
  
  render() {
    const { loading, error } = this.props;
    return (
      <div>
        <p>{loading ? 'I am loading' : 'I am NOT loading'}</p>
        <Notes/>
        { error && 
          <pre style={{ color: 'red' }}>
            {error.message 
              ? error.message 
              : error.error ? error.error : error
            }
          </pre>
        }
      </div>
    );
  }
}

export default connect(
  state => ({ 
    loading: state.loading,
    error: state.error 
  }),
  null
)(App);

