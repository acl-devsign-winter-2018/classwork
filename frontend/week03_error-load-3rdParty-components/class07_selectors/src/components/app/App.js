import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Header from './Header';
import Loading from './Loading';
import Notes from '../notes/Notes';
import Users from '../users/Users';


export default class App extends Component {
  
  render() {
    return (
      <Router>
        <div>
          <Header/>
          <main>
            <Loading/>
            <Switch>
              <Route exact path="/users" component={Users}/>
              <Route path="/users/:id/:foo" render={({ match, location, history }) => {
                console.log(match);
                return null;
              }}/>
              <Route path="/notes/:user" component={Notes}/>
              <Redirect to="/users"/>
            </Switch>
          </main>
        </div>
      </Router>
    );
  }
}
