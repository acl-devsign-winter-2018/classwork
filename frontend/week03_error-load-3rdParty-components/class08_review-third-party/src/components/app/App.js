import React, { Component, Fragment } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Albums from '../albums/Albums';
import AlbumDetail from '../albums/AlbumDetail';

class App extends Component {

  render() {
    return (
      <Router>
        <Fragment>
          <h1>Hello Actions</h1>
          <Switch>
            <Route exact path="/albums" component={Albums}/>
            <Route path="/albums/:id" component={AlbumDetail}/>
            <Redirect to="/albums"/>
          </Switch>
        </Fragment>
      </Router>
    );
  }
}

export default App;