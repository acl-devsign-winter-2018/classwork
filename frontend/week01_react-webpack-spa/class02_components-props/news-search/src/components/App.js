/* eslint react/prop-types: off */
import React, { Component } from 'react';
import './App.css';

export default class App extends Component {

  state = {
    count: 1
  };

  handleMore = () => {
    this.setState(prev => {
      return { count: prev.count + 1 };
    });
  };

  render() {
    const { count } = this.state;

    return (
      <div>
        <h1>Hello React Components</h1>
        <button onClick={this.handleMore}>More!</button>
        <A number={count}/>
      </div>
    );
  }
}

class A extends Component {

  state = {
    expanded: false
  };

  render() {
    const { expanded } = this.state;
    const { number } = this.props;

    const expansion = expanded && <B fooinator={number}/>;

    return (
      <div>
        <div>I am an instance of Component A, my number is {number}</div>
        <span onClick={() => this.setState(prev => ({ expanded: !prev.expanded }))}>
          {expanded ? 'Close' : 'Open'}
        </span>
        {expansion}
      </div>
    );
  }
}

class B extends Component {
  render() {
    return <div>I am an instance of Component B my number is {this.props.fooinator}</div>;
  }
}