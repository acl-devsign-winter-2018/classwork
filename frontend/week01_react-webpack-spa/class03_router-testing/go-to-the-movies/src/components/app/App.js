import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './App.css';

export default class App extends Component {
  static propTypes = {
    home: PropTypes.bool
  };
  
  render() {
    return (
      <div>
        { this.props.home ? <Home/> : null }
      </div>
    );
  }
}

export class Home extends Component {
  render() {
    return <div>I am the HOME page!</div>;
  }
}