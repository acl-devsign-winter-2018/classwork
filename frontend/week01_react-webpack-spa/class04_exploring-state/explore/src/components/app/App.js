import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Header from './Header';
import Game from '../game/Game';

import './App.css';

export default class App extends Component {
  
  render() {

    return (
      <Router>
        <div>
          <Header/>
          <main>
            <Switch>
              <Route exact path="/" component={Welcome}/>
              <Route path="/game" component={Game}/>
              <Redirect to="/"/>
            </Switch>
          </main>
        </div>
      </Router>
    );
  }
}