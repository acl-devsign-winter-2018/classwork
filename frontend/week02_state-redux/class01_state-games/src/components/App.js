import React, { Component } from 'react';
import './App.css';
import { rooms, start } from './rooms';
import Player from './Player';
import Room from './Room';

export default class App extends Component {

  state = {
    rooms,
    room: start,
    player: {
      name: 'player',
      inventory: []
    }
  };

  handleMove = roomKey => {
    this.setState({
      room: this.state.rooms[roomKey]
    });
  };

  handleItem = item => {
    const { room, player } = this.state;
    const index = room.items.indexOf(item);
    room.items.splice(index, 1);
    player.inventory.push(item);
    
    this.setState({
      room,
      player
    });
  };

  render() {
    const { player, room } = this.state;
    return (
      <div>
        <header>
          <h1>Shiitty Adventure</h1>
          <Player player={player}/>
        </header>
        <main>
          <Room room={room} 
            onMove={this.handleMove}
            onItem={this.handleItem}
          />
        </main>
      </div>
    );
  }
}