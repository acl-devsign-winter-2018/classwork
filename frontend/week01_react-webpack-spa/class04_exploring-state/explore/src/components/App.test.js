import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

it('passing test', () => {
  let wrapper = shallow(<App/>);
  expect(wrapper.toJSON()).toMatchSnapshot();
});