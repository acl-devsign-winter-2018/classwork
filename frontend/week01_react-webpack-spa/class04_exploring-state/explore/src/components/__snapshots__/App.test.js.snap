// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passing test 1`] = `undefined`;
